src/main/java/redis/clients/jedis/util/Pool.java

```diff
@@ -12,37 +12,20 @@
 import redis.clients.jedis.exceptions.JedisException;
 import redis.clients.jedis.exceptions.JedisExhaustedPoolException;
 
-public abstract class Pool<T> implements Closeable {
-
-  private final GenericObjectPool<T> internalPool;
+public class Pool<T> extends GenericObjectPool<T> implements Closeable {
 
   public Pool(final GenericObjectPoolConfig<T> poolConfig, PooledObjectFactory<T> factory) {
-    this.internalPool = new GenericObjectPool<>(factory, poolConfig);
+    super(factory, poolConfig);
   }
 
   @Override
   public void close() {
     destroy();
   }
 
-  public boolean isClosed() {
```

Comment:
@sazzad16 what about isClosed and clearInternalPool? Can't we keep them as delegation to avoid breaking changes too?

Reply:
@gkorland isClosed is already available in GenericObjectPool.

Reply:
@gkorland 

isClosed is already available (in super class, GenericObjectPool).

clearInternalPool is `protected`. So I took the chance of removing it in favor of an available public method `clear()` targeting a major release.